### PR TITLE
Update deprecated `QNode.(q)tape` property usage in `tests`

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

Many tests incorrectly rely on the `tape` and `qtape` properties to access information during tests. With these properties deprecated in #6583, deprecation warnings in the affected test files have been suppressed. This PR aims to replace all instances of deprecated code with the correct code.

**Description of the Change:** 

Tests should construct the tape dynamically using `construct_tape` rather than use the deprecated properties.

**Benefits:** Better testing and code maintainability.

**Possible Drawbacks:** 

I’m still a bit unsure about the changes in tests/workflow/interfaces/qnode/. The lines I deleted were originally a workaround to further verify QNode interfacing. However, the primary goal of the tests in this folder isn’t to check gradient accuracy but rather to ensure proper interfacing with the QNode architecture (e.g., verifying data types and result shapes). Since the property being used should now be considered private, relying on it in tests isn’t best practice—hence the deletion.

[sc-78317]

### CODE REVIEW

File: tests/circuit_graph/test_qasm.py [hunk 1]
```diff
@@ -349,9 +341,8 @@ def qnode():
             return qml.expval(qml.PauliZ(0))
 
         # construct the qnode circuit
-        qnode()
-
-        res = qnode.qtape.to_openqasm()
+        tape = qml.workflow.construct_tape(qnode)()
```
Comment (albi3ro) [inline tests/circuit_graph/test_qasm.py:344]:
```suggestion
        tape = qml.tape.QuantumScript([], [qml.expval(qml.Z(0))])
```

Might be bigger pain to make these changes, but did we really need a qnode and device at all?

This will make it easier to make other changes in the future.

Comment (andrijapau) [reply tests/circuit_graph/test_qasm.py:344]: Yes, good point. I will update that!

Comment (andrijapau) [reply tests/circuit_graph/test_qasm.py]: Addressed in latest commit.

File: tests/gradients/parameter_shift/test_parameter_shift_hessian.py [hunk 2]
```diff
@@ -1507,12 +1499,12 @@ def circuit(params):
...
 
         params = np.array([0.5, 0.5, 0.5], requires_grad=True)
-        circuit(params)
 
         result = qml.gradients.param_shift_hessian(circuit)(params)
         assert np.allclose(result, np.zeros((3, 3, 4)), atol=0, rtol=0)
 
-        tapes, _ = qml.gradients.param_shift_hessian(circuit.qtape)
+        tape = qml.workflow.construct_tape(circuit)(params)
+        tapes, _ = qml.gradients.param_shift_hessian(tape)
```
Comment (PietropaoloFrisoni) [inline tests/gradients/parameter_shift/test_parameter_shift_hessian.py:1507]:
```suggestion
        tapes, _ = qml.gradients.param_shift_hessian(qml.workflow.construct_tape(circuit)(params))
```

Might be worth combining into a single line (at least in this case where it's still easy to read)

File: tests/measurements/test_classical_shadow.py [hunk 3]
```diff
@@ -849,8 +839,8 @@ def test_hadamard_expval(k=1, obs=obs_hadamard, expected=expected_hadamard):
     circuit = hadamard_circuit_legacy(3, shots=50000)
     actual = circuit(obs, k=k)
 
-    print(circuit.tape)
```
Comment (PietropaoloFrisoni) [inline tests/measurements/test_classical_shadow.py:852]: Why was there a `print` statement here? XD

File: tests/workflow/interfaces/qnode/test_jax_jit_qnode.py [hunk 4]
```diff
@@ -189,10 +166,7 @@ def circuit(a, b, c):
             qml.RX(c + c**2 + jax.numpy.sin(a), wires=0)
             return qml.expval(qml.PauliZ(0))
 
-        res = jax.grad(circuit, argnums=[0, 2])(a, b, c)
-
-        if diff_method == "finite-diff":
-            assert circuit.qtape.trainable_params == [0, 2]
+        res = jax.jit(jax.grad(circuit, argnums=[0, 2]))(a, b, c)
```
Comment (PietropaoloFrisoni) [inline tests/workflow/interfaces/qnode/test_jax_jit_qnode.py:169]: So here (and below) we were not really testing `jax.jit`, although the name of the file is `test_jax_jit_qnode`. lol

**Review by PietropaoloFrisoni — COMMENTED** [review]
This looks very good to me!

About the deletion(s) in the `tests/workflow/interfaces/qnode/` file, I might miss context here, but it seems to me that you are deleting all the code lines checking properties of the `QuantumScript` (such as `trainable_params`) that it was possible to access from the `QNode` using the deprecated `tape/qtape` property (for example as `circuit.qtape.trainable_params`).

Now that the `tape/qtape` property is deprecated, we no longer have access to such properties of the `QuantumScript` from the `QNode` interface, hence the decision to remove the associated test lines. Is that correct?

File: tests/measurements/test_classical_shadow.py (hunk 3 above)
Comment (andrijapau) [reply tests/measurements/test_classical_shadow.py:852]: Lol ... oops. 😄

**Comment by andrijapau** [comment]
@PietropaoloFrisoni,

> Now that the `tape/qtape` property is deprecated, we no longer have access to such properties of the `QuantumScript` from the `QNode` interface, hence the decision to remove the associated test lines. Is that correct?

Indeed 🙂. I personally think that the deletion of these lines doesn't impact the overall test quality. More extensive testing of the ML interface gradient logic is being done in `tests/workflow/interfaces/run` and `tests/workflow/interfaces/execute`.

File: tests/gradients/parameter_shift/test_parameter_shift_hessian.py (hunk 2 above)
Comment (andrijapau) [reply tests/gradients/parameter_shift/test_parameter_shift_hessian.py]: True! Addressed in latest commit, resolving.

**Review by lillian542 — APPROVED** [review]
LGTM!
